huaweicloud/resource_huaweicloud_as_lifecycle_hook_test.go

```diff
@@ -0,0 +1,262 @@
+					resource.TestCheckResourceAttr(resourceHookName, "default_result", "CONTINUE"),
+					resource.TestCheckResourceAttr(resourceHookName, "timeout", "600"),
+					resource.TestCheckResourceAttr(resourceHookName, "notification_message",
+						"This is a update message"),
+					resource.TestCheckResourceAttr(resourceHookName, "notification_topic_urn",
+						fmt.Sprintf("urn:smn:%s:%s:update", HW_REGION_NAME, HW_PROJECT_ID)),
+				),
+			},
+			{
+				ResourceName:      resourceHookName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccASLifecycleHookImportStateIdFunc(),
+			},
+		},
+	})
+}
+
+func testAccCheckASLifecycleHookDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*config.Config)
+	asClient, err := config.AutoscalingV1Client(HW_REGION_NAME)
+	if err != nil {
+		return fmt.Errorf("Error creating huaweicloud autoscaling client: %s", err)
+	}
+
+	var groupID string
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type == "huaweicloud_as_group" {
+			groupID = rs.Primary.ID
+			continue
+		}
+	}
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "huaweicloud_as_lifecycle_hook" {
+			continue
+		}
+
+		_, err := lifecyclehooks.Get(asClient, groupID, rs.Primary.ID).Extract()
+		if err == nil {
+			return fmt.Errorf("AS lifecycle hook still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckASLifecycleHookExists(resGroup, resHook string, hook *lifecyclehooks.Hook) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resGroup]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resGroup)
+		}
+		groupID := rs.Primary.ID
+
+		rs, ok = s.RootModule().Resources[resHook]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resHook)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		config := testAccProvider.Meta().(*config.Config)
+		asClient, err := config.AutoscalingV1Client(HW_REGION_NAME)
+		if err != nil {
+			return fmt.Errorf("Error creating huaweicloud autoscaling client: %s", err)
+		}
+		found, err := lifecyclehooks.Get(asClient, groupID, rs.Primary.ID).Extract()
+		if err != nil {
+			return err
+		}
+		hook = found
+
+		return nil
+	}
+}
+
+func testAccASLifecycleHookImportStateIdFunc() resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		group, ok := s.RootModule().Resources["huaweicloud_as_group.test"]
+		if !ok {
+			return "", fmt.Errorf("Auto Scaling group not found: %s", group)
+		}
+		hook, ok := s.RootModule().Resources["huaweicloud_as_lifecycle_hook.test"]
+		if !ok {
+			return "", fmt.Errorf("Auto Scaling lifecycle hook not found: %s", hook)
+		}
+		if group.Primary.ID == "" || hook.Primary.ID == "" {
+			return "", fmt.Errorf("resource not found: %s/%s", group.Primary.ID, hook.Primary.ID)
+		}
+		return fmt.Sprintf("%s/%s", group.Primary.ID, hook.Primary.ID), nil
+	}
+}
+
+func testASLifecycleHook_base(rName string) string {
```

Comment:
can we refer to `testASV1Group_basic`?

Reply:
The topic_urn of the resource smn_topic is ForceNew, but if we want to cover `notification_topic_urn`, shall we add two smn_topic resources to the base method?